pkg/agent/attestor/node/node.go

```diff
@@ -320,13 +332,21 @@ func (a *attestor) newSVID(ctx context.Context, key *ecdsa.PrivateKey, bundle *b
 	return svid, bundle, nil
 }
 
-func (a *attestor) serverConn(ctx context.Context, bundle []*x509.Certificate) (*grpc.ClientConn, error) {
+func (a *attestor) serverConn(ctx context.Context, bundle *bundleutil.Bundle) (*grpc.ClientConn, error) {
+	var getBundle func() []*x509.Certificate
+	// If the bundle is nil, it is because we are doing an insecure bootstrap.
```

Comment:
While I see value in not adding more fields to the various Config structs, there are a few spots where this implicit judgement makes things more difficult to grok. I would much prefer to see the InsecureBootstrap bool get wired all the way through, and be used directly for decision-making purposes rather than implying its presence based on the value of bundle, trust bundle path, etc.